### PR TITLE
Remove Error logger in module-ballerina-io

### PR DESCRIPTION
**Pull request by harshalkh**
## Purpose
Fixes : https://github.com/ballerina-platform/ballerina-standard-library/issues/3083

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/DataChannelUtils.java [hunk 1]
```diff
@@ -125,8 +124,8 @@ public static Object readFloat32(BObject dataChannelObj) {
         try {
             return channel.readDouble(Representation.BIT_32);
         } catch (IOException e) {
-            log.error("error occurred while reading Float32", e);
-            return IOUtils.createError(e);
+            String message = "error occurred while reading Float32: " + e.getMessage();
```
Comment (sahanHe) [inline native/src/main/java/io/ballerina/stdlib/io/nativeimpl/DataChannelUtils.java:127]: We just need to remove the loggers. We can keep the error messages as it is

Comment (sahanHe) [reply native/src/main/java/io/ballerina/stdlib/io/nativeimpl/DataChannelUtils.java:127]: Make the necessary changes everywhere

**Comment by sahanHe** [comment]
As we are removing logs, we can also remove the 
org.slf4j.Logger;
org.slf4j.LoggerFactory;

from the imports.

We can also remove the log dependancies from [https://github.com/ballerina-platform/module-ballerina-io/blob/master/native/build.gradle]{https://github.com/ballerina-platform/module-ballerina-io/blob/master/native/build.gradle}

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/DataChannelUtils.java (hunk 1 above)
Comment (harshalkh) [reply native/src/main/java/io/ballerina/stdlib/io/nativeimpl/DataChannelUtils.java:127]: Ok

**Comment by harshalkh** [comment]
Closing this PR for now. Will reopen once I am ready with changes.

File: native/src/main/java/io/ballerina/stdlib/io/nativeimpl/DataChannelUtils.java (hunk 1 above)
Comment (harshalkh) [reply native/src/main/java/io/ballerina/stdlib/io/nativeimpl/DataChannelUtils.java]: @sahanHe changes done at all places as suggested.

**Comment by sahanHe** [comment]
> > As we are removing logs, we can also remove the org.slf4j.Logger; org.slf4j.LoggerFactory;
> > from the imports.
> > We can also remove the log dependancies from https://github.com/ballerina-platform/module-ballerina-io/blob/master/native/build.gradle
> 
> @sahanHe Imports removed wherever we can. We can't remove complete dependency as there are quite few places where we use debug logging. Do we need to remove those as well?

Hi, We don't need logs in the Java side and we can modify all the places where log4j dependancy is used

**Comment by harshalkh** [comment]
> > > As we are removing logs, we can also remove the org.slf4j.Logger; org.slf4j.LoggerFactory;
> > > from the imports.
> > > We can also remove the log dependancies from https://github.com/ballerina-platform/module-ballerina-io/blob/master/native/build.gradle
> > 
> > 
> > @sahanHe Imports removed wherever we can. We can't remove complete dependency as there are quite few places where we use debug logging. Do we need to remove those as well?
> 
> Hi @harshalkh, We don't need logs in the Java side and we can modify all the places where log4j dependancy is used

Ok working on it then.

**Comment by harshalkh** [comment]
@sahanHe I have one issue -
If I remove log object from Buffer.java then build is failing. Can you suggest -

https://github.com/harshalkh/module-ballerina-io/commit/5832ab7024cd4c6ce208ee6e25cbd52cac9273f3

https://github.com/harshalkh/module-ballerina-io/actions/runs/6704216944/job/18216218192

**Comment by harshalkh** [comment]
This PR Is ready for review @sahanHe

**Comment by sahanHe** [comment]
@daneshk Please review

**Review by sahanHe — APPROVED** [review]

**Comment by sahanHe** [comment]
@harshalkh looks good to merge. As a final addition,  can you add a log containing the issue to changelog.md file? [https://github.com/ballerina-platform/module-ballerina-io/blob/master/changelog.md](https://github.com/ballerina-platform/module-ballerina-io/blob/master/changelog.md)

**Comment by daneshk** [comment]
> LGTM. @daneshk Please review.

Please remove the entry in the [`gradle.properties`](https://github.com/ballerina-platform/module-ballerina-io/blob/master/gradle.properties#L7) file as well.

**Comment by harshalkh** [comment]
> > LGTM. @daneshk Please review.
> 
> Please remove the entry in the [`gradle.properties`](https://github.com/ballerina-platform/module-ballerina-io/blob/master/gradle.properties#L7) file as well.

Test-utils also uses this property
 We have not removed logs from test-utils.

**Comment by harshalkh** [comment]
> > > LGTM. @daneshk Please review.
> > 
> > 
> > Please remove the entry in the [`gradle.properties`](https://github.com/ballerina-platform/module-ballerina-io/blob/master/gradle.properties#L7) file as well.
> 
> Changelog updated.
> 
> Test-utils also uses this property We have not removed logs from test-utils. @daneshk @sahanHe Do I need to remove logs from tests-library as well?

@daneshk @sahanHe - I have done required changes. Removed from test-utils as well as gradle.properties.

**Review by sahanHe — APPROVED** [review]

**Comment by sahanHe** [comment]
LGTM

**Comment by harshalkh** [comment]
@daneshk pls review

**Pull request merged**